activerecord/lib/active_record/connection_adapters/postgresql/schema_creation.rb

```diff
@@ -5,15 +5,30 @@ module ConnectionAdapters
     module PostgreSQL
       class SchemaCreation < SchemaCreation # :nodoc:
         private
-          delegate :quoted_include_columns_for_index, to: :@conn
+          delegate :quoted_include_columns_for_index, :create_enum, to: :@conn
+
+          def visit_TableDefinition(o)
+            create_enums(o.columns)
+            super
+          end
 
           def visit_AlterTable(o)
+            create_enums(o.adds.map(&:column))
             sql = super
             sql << o.constraint_validations.map { |fk| visit_ValidateConstraint fk }.join(" ")
             sql << o.exclusion_constraint_adds.map { |con| visit_AddExclusionConstraint con }.join(" ")
             sql << o.unique_constraint_adds.map { |con| visit_AddUniqueConstraint con }.join(" ")
           end
 
+          def create_enums(columns)
+            return unless enums_to_create = columns.select { |c| c.type == :enum && c.options[:values] }
```

Comment:
`select` always returns an `Array` which will be a truthy value so we never return. Let's explicitly check for an `empty?` result 

```suggestion
            enums_to_create = columns.select { |c| c.type == :enum && c.options[:values] }
            return if  enums_to_create.empty?
```

Reply:
Actually, I think the whole `return` can be omitted. If `enums_to_create` is empty - `enums_to_create.each` will do nothing which is equal to returning